packages/react-core/src/demos/ComposableMenu.md

```diff
@@ -0,0 +1,665 @@
+        <Popper
+          trigger={toggle}
+          popper={menu}
+          direction="down"
+          position="left"
+          appendTo={this.containerRef.current}
+          isVisible={isExpanded}
+        />
+      </div>
+    );
+  }
+}
+```
+
+### Drilldown menu with toggle
+
+```js
+import React from 'react';
+import {
+  MenuToggle,
+  Menu,
+  MenuContent,
+  MenuList,
+  MenuItem,
+  DrilldownMenu,
+  Divider,
+  Popper
+} from '@patternfly/react-core';
+import StorageDomainIcon from '@patternfly/react-icons/dist/js/icons/storage-domain-icon';
+import CodeBranchIcon from '@patternfly/react-icons/dist/js/icons/code-branch-icon';
+import LayerGroupIcon from '@patternfly/react-icons/dist/js/icons/layer-group-icon';
+import CubeIcon from '@patternfly/react-icons/dist/js/icons/cube-icon';
+
+class DrilldownComposableMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.containerRef = React.createRef();
+    this.toggleRef = React.createRef();
+    this.menuRef = React.createRef();
+    this.state = {
+      menuDrilledIn: [],
+      drilldownPath: [],
+      menuHeights: {},
+      activeMenu: 'rootMenu',
+      isExpanded: false
+    };
+    this.onToggle = () => {
+      this.setState({
+        isExpanded: !this.state.isExpanded,
+        menuDrilledIn: [],
+        drilldownPath: [],
+        activeMenu: 'rootMenu'
+      });
+    };
+    this.drillIn = (fromMenuId, toMenuId, pathId) => {
+      this.setState({
+        menuDrilledIn: [...this.state.menuDrilledIn, fromMenuId],
+        drilldownPath: [...this.state.drilldownPath, pathId],
+        activeMenu: toMenuId
+      });
+    };
+    this.drillOut = toMenuId => {
+      const menuDrilledInSansLast = this.state.menuDrilledIn.slice(0, this.state.menuDrilledIn.length - 1);
+      const pathSansLast = this.state.drilldownPath.slice(0, this.state.drilldownPath.length - 1);
+      this.setState({
+        menuDrilledIn: menuDrilledInSansLast,
+        drilldownPath: pathSansLast,
+        activeMenu: toMenuId
+      });
+    };
+    this.setHeight = (menuId, height) => {
+      if (!this.state.menuHeights[menuId]) {
+        this.setState({
+          menuHeights: {
+            ...this.state.menuHeights,
+            [menuId]: height
+          }
+        });
+      }
+    };
+    this.handleMenuKeys = event => {
+      if ([...event.target.classList].some(c => /pf-c-menu.*/.test(c)) && this.state.isExpanded) {
+        if (event.key === 'Escape' || event.key === 'Tab') {
+          this.onToggle();
+          this.toggleRef.current.focus();
+        }
+      }
+
+      if (event.target === this.toggleRef.current && this.state.isExpanded) {
+        if (event.key === 'ArrowDown') {
+          this.menuRef.current.querySelector('button, a').focus();
+        }
+      }
+    };
+    this.handleMenuClick = event => {
+      if (![...event.target.classList].some(c => /pf-c-menu.*/.test(c)) && this.state.isExpanded) {
```

Comment:
@kmcfaul re: clicking on the SVG not toggling the menu, would this regex match with the class of the target be the culprit? The SVG doesn't have a `pf-c-menu.*` class, but if you add `pointer-events: none;` to the SVG, then your click target will be the element behind it, which is probably `.pf-c-menu-toggle__toggle-icon`

Reply:
Oh that would probably be the culprit. I'll look at adjusting the match to account for an svg target.

Reply:
Looking at it again, that function is handling clicking off the menu. The actual onClick callback for `MenuToggle` isn't working with the svg, even though the svg is contained inside the toggle `button`.

I could make the outside function handle clicking the svg as well, but I'm curious how we handled this case for Dropdown since it does work with the basic onClick. @mcoker

Reply:
Playing around with it further I believe I fixed the svg click issue. It had something to do with the event propagating. @mcoker